services/app-web/src/common-code/healthPlanFormDataType/UnlockedHealthPlanFormDataType.d.ts

```diff
@@ -66,6 +66,17 @@ type RateCapitationType = 'RATE_CELL' | 'RATE_RANGE'
 
 type ManagedCareEntity = 'MCO' | 'PIHP' | 'PAHP' | 'PCCM'
 
+type RateInfoType = {
+    rateType?: RateType
```

Comment:
🔍 I would like to take another look at types here. A couple questions

- I notice that for both unlocked and locked we are adding additional types but still keeping the old rate related types as well. That's double the types. Why aren't we removing the types we won't use anymore? 
-  Is there a reason to give `RateInfoType` such a loose type with less optional fields? Stronger typing could better reflects its state in locked submission (with required fields where we expect them to be present). The rateInfos used in unlockedSubmissions seem to be to be `Partial<RateInfoType>`.

Reply:
> I notice that for both unlocked and locked we are adding additional types (rateInfos) but still keeping the old rate related types as well. That's double the types. Why aren't we removing the types we won't use anymore? I was assuming we would remove the old fields since we will be relying on the array going forward. Let me know what you are thinking.

The review and summary pages still use the old fields. Switching them over is a part of another task [MR-2527](https://qmacbis.atlassian.net/browse/MR-2527). But I think I could get the pages to use the first rate from `rateInfos` for now and leave the UI work for MR-2527.

> Is there a reason to give RateInfoType such a loose type with less optional fields? Stronger typing could better reflects its state in locked submission (with required fields where we expect them to be present). The rateInfos used in unlockedSubmissions seem to be to be Partial<RateInfoType>.

That's a good point, I'll make those changes. I'm wondering if that would cause issues with save as draft, but I'll test it out.

Reply:
@haworku There were more Backend work than I initially thought. So there will be more changes coming.

Reply:
Sounds good no worries. I know this PR was still in draft and not yet in review, thought I would take a look. Let me know if there's anything you want to pair on as well. I think we underestimated this ticket, this is a significant chunk of work you took on, thanks for getting things in order!